### DISCUSSION TITLE
Zig Style Switch

### DISCUSSION THREAD

**Discussion by jpx40** [c3lang/c3c#1394] [Ideas]
Hey, 
i wanted to ask if an modern way of writing Switch Statement like in zig would be a great idea for c3c

ZIg Style switch:
```
   var x: i8 = 10;
    switch (x) {
        -1...1 => {
            x = -x;
        },
        10, 100 => {
            //special considerations must be made
            //when dividing signed integers
            x = @divExact(x, 10);
        },
        else => {},
    }
    try expect(x == 1);
```

**Comment by lerno** [comment]
The corresponding C3 code:
```c
char x = 10;
switch (x)
{
  case -1..1: x = -x;
  case 10: case 100: x = x / 10;
}
assert(x == 1)
```
Is there any advantage to the Zig code, which is longer and more verbose?

**Reply by jpx40** [reply]
i don't think so it's just preference of me sry

**Reply by lerno** [reply]
The design constraint of C3 is to stick to C syntax and semantic unless there is a huge benefit departing from it. In the case of the C3 switch case, its functionality has been extended, and it has implicit breaks. It also has a nextcase statement for which there is no counterpart in Zig. Despite this, the overall semantics and syntax is familiar to C programmers. Arbitrarily changing the syntax from C for no benefit, violates the design constraints.

**Reply by jpx40** [reply]
Kinda unrleated question are strings 0 terminated in c3?

**Reply by lerno** [reply]
Short answer, the normal `String` isn't (it is a distinct char slice). There is a ZString for C-style zero terminated strings for interop with libraries using C style strings.

**Reply by jpx40** [reply]
ok thanks
